apcd-cms/src/apps/utils/static/utils/js/checkForBlankInputs.js

```diff
@@ -1,7 +1,11 @@
 function noEmptyInputs(inputArr) {
   inputArr.forEach(i => i.addEventListener('change', () => {
-      if (i.value.replace(/ /g, '') === '') { 
-        i.value = ''; // if input is all spaces, want to clear it out to leverage <input>'s required prop
-      }
+    const cleanedValue = i.value.replace(/[^a-zA-Z0-9\s-@!?,%$#&*().]|<script>|<\/script>|'|"|}|{|javascript|JavaScript|\.exe/gi, '') // Allow specified characters
```

Comment:
Have we thought about tools like [DOMPurify](https://github.com/cure53/DOMPurify)? instead of us writing a regex and maintaining this?

Reply:
Thanks for working with me this morning on this! Great solution. I will get with @wesleyboar about possibly adding package to Core-CMS. I created the follow on task here. https://jira.tacc.utexas.edu/browse/WP-240

Reply:
Excellent suggestion. I'll evaluate how it could be put in Core.[^1]

I think you can use it in APCD-CMS as is, without waiting on Core.

<details><summary>Do We Use External JavaScript like that in a TACC CMS?</summary>

_**Yes.** [BrainMap](https://portal.brainmap.org) is one of several sites with [a snippet that loads and uses external JavaScript](https://portal.brainmap.org/admin/djangocms_snippet/snippet/4/change/). The snippet imports JavaScript form an external CDN, then uses it to manipulate the DOM._

```
<script type="module" id="js-ext-link-open-new-tab">
    import findLinksAndSetTargets from 'https://cdn.jsdelivr.net/gh/TACC/Core-CMS@v3.11.0-alpha.1/taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js';

    /* To make certain links open in new window, and apply a11y attribute */
    findLinksAndSetTargets({
        pathsToForceSetTarget: [
            '/tools/sleuth/',
            '/tools/gingerale/',
            '/tools/mango/',
            '/guides/brainmap-cli-hot-guide-redirect/'
        ]
    });
</script>
```

</details>

<details><summary>How Can I Use this on APCD CMS Right Now?</summary>

_I have enough immediate bandwidth only to show you how to **load** it, and **begin to use** it. Please attempt to implement it on APCD CMS as needed. That can teach me how Core can provide it for a client.[^1]_

**Local Usage**

Local Load & Usage
```
<script type="module" id="js-purify-dom">
	import * as DOMPurify from 'dompurify';

	const specificFieldDataA = ''/* ... */;
	const cleanedFieldDataA = DOMPurify.sanitize(specificFieldDataA);

    /* ... */
</script>
```

**Global Usage**

Global Load
```
<script src="dist/purify.min.js"></script>
```

Local Use of Global Load
```
<script>
	const specificFieldDataA = ''/* ... */;
	const cleanedFieldDataA = DOMPurify.sanitize(specificFieldDataA);

    /* ... */
</script>
```

</details>

[^1]: (Warning: Train of Thought) I could make it load on pages that use Django CMS Form, and sanitize all(?) field input blindly, or limit it to only fields with a certain name. But then I must rely on CMS admins to create forms using the "correct" names, but policing that is overhead, and overhead I cannot automate. Attempting to integrate with Django CMS Form also worries me, because that outdated and barely-maintained plugin is one reason we can't quickly update to latest version of Django...

Reply:
Edited message. I fixed string typo. Basically, it can be loaded as module from CDN via https://cdn.jsdelivr.net/gh/cure53/DOMPurify@3.0.5/dist/purify.js.